src/main/java/io/supertokens/session/Session.java

```diff
@@ -669,59 +671,144 @@ public static String[] revokeSessionUsingSessionHandles(Main main,
-                    .getSession(tenantIdentifierWithStorage, sessionHandle) == null) {
-                result[indexIntoResult] = sessionHandle;
-                indexIntoResult++;
+            sessionHandleMap.get(tenantId).add(sessionHandle);
+        }
+
+        int numberOfSessionsRevoked;
+        List<String> revokedSessionHandles = new ArrayList<>();
+
+        for (String tenantId : sessionHandleMap.keySet()) {
+            String[] sessionHandlesForTenant = sessionHandleMap.get(tenantId).toArray(new String[0]);
+
+            TenantIdentifier tenantIdentifier = new TenantIdentifier(appIdentifierWithStorage.getConnectionUriDomain(), appIdentifierWithStorage.getAppId(), tenantId);
+            TenantIdentifierWithStorage tenantIdentifierWithStorage = null;
+            try {
+                tenantIdentifierWithStorage = tenantIdentifier.withStorage(
+                        StorageLayer.getStorage(tenantIdentifier, main));
+            } catch (TenantOrAppNotFoundException e) {
+                // ignore as this can happen if the tenant has been deleted after fetching the sessionHandles
+                continue;
             }
+
+            String[] sessionHandlesRevokedForTenant = revokeSessionUsingSessionHandles(tenantIdentifierWithStorage,
+                    sessionHandlesForTenant);
+            revokedSessionHandles.addAll(Arrays.asList(sessionHandlesRevokedForTenant));
         }
 
-        return result;
+        return revokedSessionHandles.toArray(new String[0]);
+    }
+
+    private static String[] revokeSessionUsingSessionHandles(TenantIdentifierWithStorage tenantIdentifierWithStorage,
+                                                            String[] sessionHandles)
+            throws StorageQueryException {
+        Set<String> validHandles = new HashSet<>();
+
+        if (sessionHandles.length > 1) {
+            // we need to identify which sessionHandles are valid if there are more than one sessionHandles to revoke
+            // if there is only one sessionHandle to revoke, we would know if it was valid by the number of revoked sessions
+            for (String sessionHandle : sessionHandles) {
+                if (tenantIdentifierWithStorage.getSessionStorage()
+                        .getSession(tenantIdentifierWithStorage, sessionHandle) != null) {
+                    validHandles.add(sessionHandle);
+                }
+            }
+        }
+
+        int numberOfSessionsRevoked = tenantIdentifierWithStorage.getSessionStorage()
+                .deleteSession(tenantIdentifierWithStorage, sessionHandles);
+
+        // most of the time we will enter the below if statement
+        if (numberOfSessionsRevoked == sessionHandles.length) {
+            return sessionHandles;
+        } else if (numberOfSessionsRevoked == 0) {
+            return new String[0];
+        } else {
+            String revokedSessionHandles[] = new String[numberOfSessionsRevoked];
```

Comment:
make it an array list and then return the to array of that at the end

Reply:
Done